yjit/src/backend/arm64/mod.rs

```diff
@@ -1588,4 +1603,16 @@ mod tests {
             0x0: ldur x1, [x19, #8]
         "});
     }
+
+    #[test]
+    fn test_not_split_mov() {
+        let (mut asm, mut cb) = setup_asm();
+
+        asm.mov(Opnd::Reg(Assembler::TEMP_REGS[0]), Opnd::UImm(5));
```

Comment:
I would test with 0xffff and 0x10000 to show that it handles the boundary condition properly.